src/server_manager/web_app/app.ts

```diff
@@ -1206,29 +1208,31 @@ export class App {
     });
   }
 
-  private setMetricsEnabled(metricsEnabled: boolean) {
-    this.selectedServer.setMetricsEnabled(metricsEnabled)
-        .then(() => {
-          // Change metricsEnabled property on polymer element to update display.
-          this.appRoot.getServerView(this.appRoot.selectedServer.id).metricsEnabled =
-              metricsEnabled;
-        })
-        .catch((error) => {
-          console.error(`Failed to set metrics enabled: ${error}`);
-          this.appRoot.showError(this.appRoot.localize('error-metrics'));
-        });
+  private async setMetricsEnabled(metricsEnabled: boolean) {
+    try {
+      this.selectedServer.setMetricsEnabled(metricsEnabled);
```

Comment:
Isn't this call asynchronous? Don't you need to wait?

Reply:
Good catch, fixed in 97d7c32.